src/data/UserD2Repository.ts

```diff
@@ -63,4 +67,78 @@ export class UserD2Repository implements UserRepository {
             .compact()
             .value();
     }
+
+    async saveAll(users: User[], from: UserAttribute, to: UserAttribute): Async<MigrationResult> {
```

Comment:
The repo should be plain and simple, the from/to logic should be in useCases/entities: 

`async saveAll(users: User[]): Async<Stats>`